app/assets/javascripts/events/edit_event_form.js

```diff
@@ -0,0 +1,35 @@
+var editEventForm = {
+    timezone : function() {
+        return timeZoneUtilities.detectUserTimeZone();
+    },
+    handleUserTimeZone: function () {
+        if ($("#start_time_tz").length) {
+            $('#start_time_tz').setToUserTimeZone(this.timezone());
+        }
+        if ($('#start_time').length && $('#start_date').length) {
+            this.updateDateAndTimeToUserTimeZone($('#next_date'), $('#start_date'), $('#start_time'));
```

Comment:
May be there is some better abstraction. It looks like current implementation uses DOM as storage, some data stored at DOM objects (#next_data). If this function will be execute from several places each time DOM tree will be traversed multiple times in order to find this data stored in it. If tree is large there can be some performance issues. May it possible to have some model and retrieve data from there?  May be it's possible to use Local storage for that case https://developer.mozilla.org/en/docs/Web/API/Window/localStorage